fix: conditional sibling islands DOM order

marvinhagemeister opened the pull request:
This happened when both share the same parent and one or both render nothing into the DOM. Once an island starts rendering something into our root fragment that we wrap around each island, it was guaranteed to call `.appendChild()`. But when islands share the same parent this leads to children being added to the very end of the parent DOM node instead of just of the end of the current island tree.

For that reason we need to keep at least the end comment marker of an island as a reference point of where `appendChild` should insert children.

Fixes the first issue mentioned in #1322

lucacasonato reviewed (commented):
Great! Can you try the empty text node approach? If too complicate, this is fine too.

This doesn't fix the issue if the conditional element is `props.children`, right?

marvinhagemeister committed:
fix: conditional sibling islands DOM order

This happened when both share the same parent and one or both render
nothing into the DOM.

lucacasonato reviewed (approved):
LGTM
